extensions/interactions/CodeRepl/directives/oppia-short-response-code-repl.component.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the CodeRepl short response.
+ *
+ * IMPORTANT NOTE: The naming convention for customization args that are passed
+ * into the directive is: the name of the parameter, followed by 'With',
+ * followed by the name of the arg.
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { HtmlEscaperService } from 'services/html-escaper.service';
+
+@Component({
+  selector: 'oppia-short-response-code-repl',
+  templateUrl: './code-repl-short-response.component.html',
+  styleUrls: []
+})
+export class ShortResponseCodeRepl implements OnInit {
+  @Input('answer') answerWithValue: string;
+  answer;
```

Comment:
What does this do?

Reply:
Basically in the template you bind to answer, in ts file we can change what variable the value will be passed to. Then usage notes here might be helpful: https://angular.io/api/core/Input